src/V3Coverage.cpp

```diff
@@ -542,6 +577,230 @@ class CoverageVisitor final : public VNVisitor {
         lineTrack(nodep);
     }
 
+    void finalizeExprComments() {
+        for (CoverExpr& expr : m_exprs) {
+            UASSERT(m_seeking != NONE, "Finalizing expression comments without an objective");
+            expr.m_comment = "(" + expr.m_comment + ") => " + (m_objective ? "1" : "0");
+        }
+    }
+
+    bool checkMaxExprs() {
+        if (m_seeking == ABORTED) { return true; }
+        if (static_cast<int>(m_exprs.size()) <= v3Global.opt.coverageExprMax()) { return false; }
+
+        m_seeking = ABORTED;
+        for (CoverExpr& expr : m_exprs) { expr.m_expr->deleteTree(); }
+        m_exprs.clear();
+        return true;
+    }
+
+    void coverExprs(AstNodeExpr* nodep) {
+        if (!m_state.exprCoverageOn(nodep) || nodep->dtypep()->width() != 1 || m_proc == nullptr) {
+            return;
+        }
+
+        VL_RESTORER(m_seeking);
+        VL_RESTORER(m_objective);
+        FileLine* fl = nodep->fileline();
+
+        m_seeking = SEEKING;
+        m_exprs.clear();
+        m_objective = false;
+        iterate(nodep);
+        finalizeExprComments();
+        CoverExprs falseExprs;
+        m_exprs.swap(falseExprs);
+
+        m_objective = true;
+        iterate(nodep);
+        finalizeExprComments();
+        m_exprs.splice(m_exprs.end(), falseExprs);
+        checkMaxExprs();
+
+        int count = 0;
+        for (CoverExpr& expr : m_exprs) {
+            string name = "expr_" + std::to_string(count);
+            AstNode* const newp = newCoverInc(fl, "", "v_expr", expr.m_comment, "", 0,
+                                              traceNameForLine(nodep, name));
+            AstIf* ifp = new AstIf(fl, expr.m_expr, newp, nullptr);
+            ifp->user2(true);
+            insertProcStatement(m_proc, ifp);
+            count++;
```

Comment:
```suggestion
            const string name = "expr_" + std::to_string(count);
            AstNode* const newp = newCoverInc(fl, "", "v_expr", expr.m_comment, "", 0,
                                              traceNameForLine(nodep, name));
            AstIf* const ifp = new AstIf(fl, expr.m_expr, newp, nullptr);
            ifp->user2(true);
            insertProcStatement(m_proc, ifp);
            ++count;
```

Many other missing consts - check all please.